File: src/my_robot_tutorials/scripts/multiply_two_numbers_client.py
#!/usr/bin/env python3

import sys

import rospy
from my_robot_tutorials.srv import MultiplyTwoNumbers, MultiplyTwoNumbersResponse


SERVICE_NAME = "/multiply_two_numbers"

def main(a: float, b: float):
    rospy.wait_for_service(SERVICE_NAME)
    print(f"Requesting {a} * {b}...")

    try:
        s = rospy.ServiceProxy(SERVICE_NAME, MultiplyTwoNumbers)
        response: MultiplyTwoNumbersResponse = s.call(a, b)
        print(f"{a} * {b} = {response.result}")
    except rospy.ServiceException as e:
        print(f"Service call failed: {e}")


def usage():
    return f"{sys.argv[0]} a b"


if __name__ == "__main__":
    if len(sys.argv) == 3:
        a = float(sys.argv[1])
        b = float(sys.argv[2])
    else:
        print(usage())
        sys.exit(1)

    main(a, b)

File: src/my_robot_tutorials/src/multiply_two_numbers_client.cpp
#include <string>

#include <ros/ros.h>
#include "my_robot_tutorials/MultiplyTwoNumbers.h"

const std::string SERVICE_NAME = "/multiply_two_numbers";

std::string usage(char **argv) {
    return std::string(argv[0]) + " a b";
}

int main(int argc, char **argv) {
    ros::init(argc, argv, "multiply_two_numbers_client");
    auto node_handle = ros::NodeHandle();

    if (argc != 3) {
        ROS_ERROR("%s", usage(argv).c_str());
        std::exit(1);
    }

    ros::service::waitForService(SERVICE_NAME);

    auto client = node_handle.serviceClient<my_robot_tutorials::MultiplyTwoNumbers>(SERVICE_NAME);

    auto srv = my_robot_tutorials::MultiplyTwoNumbers();
    srv.request.a = atof(argv[1]);
    srv.request.b = atof(argv[2]);
    ROS_INFO("Requesting %.3lf * %.3lf...", srv.request.a, srv.request.b);

    if (client.call(srv)) {
        ROS_INFO("%.3lf * %.3lf = %.3lf", srv.request.a, srv.request.b, srv.response.result);
    } else {
        ROS_ERROR("Service call failed");
    }
}
